src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -492,6 +493,39 @@ def _build_storage_profile():
             storage_profiles['CommunityGalleryImage']['osDisk']['managedDisk']['diskEncryptionSet'] = {
                 'id': os_disk_encryption_set,
             }
+        if os_disk_security_encryption_type is not None:
+            storage_profiles['ManagedPirImage']['osDisk']['managedDisk'].update({
+                'securityProfile': {
+                    'securityEncryptionType': os_disk_security_encryption_type,
+                    'diskEncryptionSet': {
+                        'id': os_disk_securevm_disk_encryption_set
+                    }
+                }
+            })
```

Comment:
Please note that `--os-disk-securevm-disk-encryption-set` can only be passed in when `--os-disk-security-encryption-type` is `DiskwithVMGuestState`, otherwise this property cannot be sent to the service